src/bom/bitcode.rs

```diff
@@ -637,59 +594,41 @@ fn build_bitcode_tar(bc_target : &OsString,
 /// We pass in the working directory in which the objects were constructed so
 /// that we can generate appropriate commands (in terms of absolute paths) so
 /// that we don't need to worry about where we are replaying the build from.
-fn attach_bitcode(chan : &mut mpsc::Sender<Option<Event>>,
-                  cwd : &Path,
-                  orig_target : &OsString,
+fn attach_bitcode(cwd : &Path,
+                  bc_args : &mut BitcodeArguments,
                   bc_target : &OsString) -> anyhow::Result<()> {
-    let object_path = to_absolute(cwd, orig_target);
-    let bc_path = to_absolute(cwd, bc_target);
-
-    let mut hasher = Sha256::new();
-    let mut bc_content = Vec::new();
-    let mut bc_file = std::fs::File::open(&bc_path)?;
-    bc_file.read_to_end(&mut bc_content)?;
-    hasher.update(bc_content);
-    let hash = hasher.finalize();
-
-    // This temporary file is filled in by `build_bitcode_tar`; however, we
-    // allocate it here so that it is still in scope (i.e., not deleted) when we
-    // invoke `inject_bitcode`
-    let mut tar_file = tempfile::NamedTempFile::new()?;
-    build_bitcode_tar(bc_target, bc_path.as_path(), hash.as_slice(), &mut tar_file)?;
-
-    let tar_name = OsString::from(tar_file.path());
-    let ok_tar_name = tar_name.into_string().ok().unwrap();
-    let opath = object_path.into_os_string();
-    inject_bitcode(chan, cwd, orig_target, bc_target, &ok_tar_name, &opath)
-}
-
-pub const ELF_SECTION_NAME : &str = ".llvm_bitcode";
+    let mktar = bc_args.ops.push_op(
+        SubProcOperation::calling(
+            |_cwd, args|
+            build_bitcode_tar(&args[0],
+                              &PathBuf::from(args[2].clone()).as_path(),
+                              &args[1],
+                              File::create(PathBuf::from(args[3].clone()).as_path())?)));
+    // Define the args in the order that matches the args in the calling() above.
+    mktar.set_input(&FileSpec::Append(NamedFile::temp("")));
+    mktar.set_output(&FileSpec::Append(NamedFile::temp(".tar")));
+    let mut reprname = bc_args.ops.out_file_for_chain().unwrap_or(PathBuf::from("unk.bc"));
```

Comment:
Getting pedantic, but I recommend `unwrap_or_else` to lazily construct the default.

Reply:
Good catch.  Updated.